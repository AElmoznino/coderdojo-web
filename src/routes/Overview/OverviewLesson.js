// @flow

import React from 'react'
import styled from 'styled-components'
import ButtonLink from 'components/Button/ButtonLink'
import H4 from 'components/Typography/H4'

type OverviewLessonProps = {
  lessonNumber: number,
  image: {
    alt: string,
    url: string,
  },
  text: string,
}

const Wrap = styled.div`
  background-color: ${({ theme }) => theme.colors.white};
  padding: ${({
    theme: {
      sizes: { medium, third },
    },
  }) => `${third} ${third} ${medium}`};
`

export const OverviewLesson = ({
  lessonNumber,
  image,
  text,
}: OverviewLessonProps) => (
  <Wrap>
    <H4>Lektion {lessonNumber}</H4>
    <img alt={image.alt} src={image.url} />
    <p>{text}</p>
    <ButtonLink to={`/lesson/${lessonNumber}`}>Kom ig√•ng</ButtonLink>
  </Wrap>
)

export default OverviewLesson
