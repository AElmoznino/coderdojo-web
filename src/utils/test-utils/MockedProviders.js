import * as React from 'react'
import { ApolloProvider } from 'react-apollo'
import { BrowserRouter } from 'react-router-dom'
import { ThemeProvider } from 'styled-components'
import { theme } from '../../theme'
import ApolloClient from 'apollo-boost'
import createMockedNetworkFetch from 'apollo-mocknetworkinterface'

jest.unmock('react-apollo')
jest.unmock('react-router-dom')

type MockedProvidersProps = {
  children: React.Node,
  loading?: boolean,
  response?: Object,
}

const MockedProviders = ({
  children,
  loading,
  response = {},
}: MockedProvidersProps) => {
  const createResponse = () => {
    return {
      loading: !!loading,
      data: {
        ...response,
      },
    }
  }

  const mockedNetworkFetch = createMockedNetworkFetch(createResponse, {
    latency: 0,
  })

  const client = new ApolloClient({
    uri: 'http://localhost:4000',
    fetch: mockedNetworkFetch,
  })

  return (
    <ApolloProvider client={client}>
      <ThemeProvider theme={theme}>
        <BrowserRouter>{children}</BrowserRouter>
      </ThemeProvider>
    </ApolloProvider>
  )
}

export default MockedProviders
