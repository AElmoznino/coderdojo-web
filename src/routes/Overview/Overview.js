// @flow

import React from 'react'
import styled from 'styled-components'
import { Query } from 'react-apollo'
import gql from 'graphql-tag'
import type { Match } from 'react-router-dom'
import ReactMarkdown from 'react-markdown'
import ErrorMessage from 'components/ErrorMessage/ErrorMessage'
import GridColumn from 'components/Grid/GridColumn'
import GridSection from 'components/Grid/GridSection'
import H1 from 'components/Typography/H1'
import LoadingIndicator from 'components/LoadingIndicator/LoadingIndicator'
import OverviewLesson from './OverviewLesson'

type OverviewProps = {
  match: Match,
}

const LessonsWrap = styled.div`
  display: grid;
  grid-template-columns: repeat(4, 200px);
  grid-column-gap: ${({ theme }) => theme.sizes.medium};
`

const GET_OVERVIEW = gql`
  query getOverview($level: String) {
    overview(level: $level) {
      description
      lessons {
        lessonId
        image {
          alt
          url
        }
        text
        title
      }
    }
  }
`

const Overview = ({ match }: OverviewProps) => {
  return (
    <Query query={GET_OVERVIEW} variables={{ level: match.params.level }}>
      {({ data: { overview }, error, loading }) => {
        if (loading) return <LoadingIndicator />
        if (error) return <ErrorMessage />
        if (!overview) return null

        return (
          <GridColumn>
            <GridSection>
              <H1>Ã–versikt av {match.params.level}</H1>
              <ReactMarkdown>{overview.description}</ReactMarkdown>
              <LessonsWrap>
                {overview.lessons.map((lesson, i) => (
                  <OverviewLesson
                    key={lesson.lessonId}
                    lessonNumber={i + 1}
                    {...lesson}
                  />
                ))}
              </LessonsWrap>
            </GridSection>
          </GridColumn>
        )
      }}
    </Query>
  )
}

export default Overview
