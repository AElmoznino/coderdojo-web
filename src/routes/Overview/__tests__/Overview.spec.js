import React from 'react'
import { mount } from 'enzyme'
import Overview from '../Overview'
import MockedProviders from 'utils/test-utils/MockedProviders'

jest.unmock('graphql-tag')
jest.unmock('react-apollo')

describe('components/Overview', () => {
  let component
  let match
  let mockedResponse

  beforeEach(() => {
    match = {
      params: {
        level: 'nybörjare',
      },
    }

    mockedResponse = {
      overview: {
        __typename: 'Overview',
        description:
          'Har du aldrig kodat förrut? Då börjar du här och får en\n                introduktion till hur man strukturerar upp hemsidor med **HTML** och\n                snyggar till dem med **CSS**.',
        lessons: [
          {
            __typename: 'LessonOverview',
            lessonId: 'nybörjare-1',
            image: {
              __typename: 'Image',
              alt: 'HTML för nybörjare, del 1',
              url: 'https://placeimg.com/180/180/tech',
            },
            text: 'Här får du lära dig grunderna i HTML',
            title: 'HTML del 1',
          },
          {
            __typename: 'LessonOverview',
            lessonId: 'nybörjare-2',
            image: {
              __typename: 'Image',
              alt: 'HTML för nybörjare, del 2',
              url: 'https://placeimg.com/180/180/tech',
            },
            text: 'Här får du lära dig mer om HTML',
            title: 'HTML del 2',
          },
          {
            __typename: 'LessonOverview',
            lessonId: 'nybörjare-3',
            image: {
              __typename: 'Image',
              alt: 'CSS för nybörjare',
              url: 'https://placeimg.com/180/180/tech',
            },
            text: 'Här får du lära dig grunderna i CSS',
            title: 'CSS del 1',
          },
          {
            __typename: 'LessonOverview',
            lessonId: 'nybörjare-4',
            image: {
              __typename: 'Image',
              alt: 'CSS för nybörjare',
              url: 'https://placeimg.com/180/180/tech',
            },
            text: 'Här får du lära dig mer om CSS',
            title: 'CSS del 2',
          },
        ],
      },
    }

    component = mount(
      <MockedProviders response={mockedResponse}>
        <Overview match={match} />
      </MockedProviders>,
    )
  })

  it('renders Overview', done => {
    setImmediate(() => {
      component.update()

      expect(component).toMatchSnapshot()

      done()
    })
  })
})
