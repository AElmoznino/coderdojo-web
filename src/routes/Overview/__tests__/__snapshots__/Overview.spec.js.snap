// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Overview renders Overview 1`] = `
<styled.div>
  <div
    className="sc-bdVaJa cBlAJt"
  >
    <styled.div>
      <div
        className="sc-bwzfXH LKvyz"
      >
        <styled.h1>
          <h1
            className="sc-htpNat kpiFDp"
          >
            Översikt av 
            nybörjare
          </h1>
        </styled.h1>
        <p
          key="paragraph-1-1"
        >
          Har du aldrig kodat förrut? Då börjar du här och får en
                introduktion till hur man strukturerar upp hemsidor med 
          <strong
            key="strong-2-73"
          >
            HTML
          </strong>
           och
                snyggar till dem med 
          <strong
            key="strong-3-38"
          >
            CSS
          </strong>
          .
        </p>
        <styled.div>
          <div
            className="sc-htoDjs iAmKHR"
          >
            <styled.div>
              <div
                className="sc-gzVnrw gINtiU"
              >
                <styled.h4>
                  <h4
                    className="sc-bZQynM FsPVD"
                  >
                    Lektion 
                    1
                  </h4>
                </styled.h4>
                <img
                  alt="HTML för nybörjare, del 1"
                  src="https://placeimg.com/180/180/tech"
                />
                <p>
                  Här får du lära dig grunderna i HTML
                </p>
                <Styled(Link)
                  to="/lesson/nybörjare-1"
                >
                  <a
                    className="sc-EHOje fwWOIl"
                    href="/lesson/nybörjare-1"
                    onClick={[Function]}
                  >
                    Kom igång
                  </a>
                </Styled(Link)>
              </div>
            </styled.div>
            <styled.div>
              <div
                className="sc-gzVnrw gINtiU"
              >
                <styled.h4>
                  <h4
                    className="sc-bZQynM FsPVD"
                  >
                    Lektion 
                    2
                  </h4>
                </styled.h4>
                <img
                  alt="HTML för nybörjare, del 2"
                  src="https://placeimg.com/180/180/tech"
                />
                <p>
                  Här får du lära dig mer om HTML
                </p>
                <Styled(Link)
                  to="/lesson/nybörjare-2"
                >
                  <a
                    className="sc-EHOje fwWOIl"
                    href="/lesson/nybörjare-2"
                    onClick={[Function]}
                  >
                    Kom igång
                  </a>
                </Styled(Link)>
              </div>
            </styled.div>
            <styled.div>
              <div
                className="sc-gzVnrw gINtiU"
              >
                <styled.h4>
                  <h4
                    className="sc-bZQynM FsPVD"
                  >
                    Lektion 
                    3
                  </h4>
                </styled.h4>
                <img
                  alt="CSS för nybörjare"
                  src="https://placeimg.com/180/180/tech"
                />
                <p>
                  Här får du lära dig grunderna i CSS
                </p>
                <Styled(Link)
                  to="/lesson/nybörjare-3"
                >
                  <a
                    className="sc-EHOje fwWOIl"
                    href="/lesson/nybörjare-3"
                    onClick={[Function]}
                  >
                    Kom igång
                  </a>
                </Styled(Link)>
              </div>
            </styled.div>
            <styled.div>
              <div
                className="sc-gzVnrw gINtiU"
              >
                <styled.h4>
                  <h4
                    className="sc-bZQynM FsPVD"
                  >
                    Lektion 
                    4
                  </h4>
                </styled.h4>
                <img
                  alt="CSS för nybörjare"
                  src="https://placeimg.com/180/180/tech"
                />
                <p>
                  Här får du lära dig mer om CSS
                </p>
                <Styled(Link)
                  to="/lesson/nybörjare-4"
                >
                  <a
                    className="sc-EHOje fwWOIl"
                    href="/lesson/nybörjare-4"
                    onClick={[Function]}
                  >
                    Kom igång
                  </a>
                </Styled(Link)>
              </div>
            </styled.div>
          </div>
        </styled.div>
      </div>
    </styled.div>
  </div>
</styled.div>
`;
